pinot-core/src/test/java/org/apache/pinot/core/util/TableConfigUtilsTest.java

```diff
@@ -1098,4 +1105,46 @@ public void testValidateUpsertConfig() {
       Assert.assertEquals(e.getMessage(), "The upsert table cannot have star-tree index.");
     }
   }
+
+  @Test
+  public void testValidatePartialUpsertConfig() {
+    Schema schema =
+        new Schema.SchemaBuilder().setSchemaName(TABLE_NAME).addSingleValueDimension("myCol1", FieldSpec.DataType.LONG)
+            .addSingleValueDimension("myCol2", FieldSpec.DataType.STRING)
+            .setPrimaryKeyColumns(Lists.newArrayList("myCol1")).build();
+
+    Map<String, String> streamConfigs = new HashMap<>();
+    streamConfigs.put("stream.kafka.consumer.type", "highLevel");
+    streamConfigs.put("streamType", "kafka");
+    streamConfigs.put("stream.kafka.topic.name", "test");
+    streamConfigs
+        .put("stream.kafka.decoder.class.name", "org.apache.pinot.plugin.stream.kafka.KafkaJSONMessageDecoder");
+    streamConfigs.put("stream.kafka.consumer.type", "simple");
+
+    TableConfig tableConfig = new TableConfigBuilder(TableType.REALTIME).setTableName(TABLE_NAME)
+        .setUpsertConfig(new UpsertConfig(UpsertConfig.Mode.PARTIAL)).setNullHandlingEnabled(false)
+        .setRoutingConfig(new RoutingConfig(null, null, RoutingConfig.STRICT_REPLICA_GROUP_INSTANCE_SELECTOR_TYPE))
+        .setStreamConfigs(streamConfigs).build();
+    try {
+      TableConfigUtils.validateUpsertConfig(tableConfig, schema);
+    } catch (Exception e) {
+      Assert.assertEquals(e.getMessage(), "NullValueHandling is required to be enabled for partial upsert tables.");
+    }
+
+    Map<String, UpsertConfig.Strategy> partialUpsertStratgies = new HashMap<>();
+    partialUpsertStratgies.put("myCol1", UpsertConfig.Strategy.INCREMENT);
+    try {
+      TableConfigUtils.validateUpsertConfig(tableConfig, schema);
+    } catch (Exception e) {
+      Assert.assertEquals(e.getMessage(), "INCREMENT merger cannot be applied to PK.");
+    }
+
+    partialUpsertStratgies = new HashMap<>();
+    partialUpsertStratgies.put("myCol2", UpsertConfig.Strategy.INCREMENT);
+    try {
+      TableConfigUtils.validateUpsertConfig(tableConfig, schema);
+    } catch (Exception e) {
+      Assert.assertEquals(e.getMessage(), "INCREMENT merger should be numeric data types.");
```

Comment:
Thanks @yupeng9 for pointing this out. I don't see a use case that requires increment on timestamp for now. Please let me know there is any use case require this.
Since time type can be Numeric Datatype, I added one more validation rules to avoid Numeric timestamp value get incremented.

Reply:
One possible use case is the time counter increments.

Reply:
I don't think it makes much sense to have an increment time column:
1. Main time column cannot be changed because it is used to manage the upsert events
2. Time column should follow the configured time format
3. I don't see a use case where user would put the delta value for time instead of the actual value. It is hard to manage a flow to have the initial value as the first event

Reply:
hmm, one example (contrived) is the driver/rider wait time, which can increment periodically (e.g. +5 min wait time)

Reply:
@yupeng9 The wait time is not really a time column, but a dimension column. Time column must be timestamp or datetime. There is no limit on the data type, as long as the value is stored as numeric type

Reply:
I see. By time column, I meant a dimension column of `Timestamp` type, which is added recently? Will the check fail this type?

Reply:
It won't fail, but still we should not try to increment the timestamp because there is no guarantee that the base value is already ingested (or very hard to manage at least)

Reply:
I see. So in this case, it's better to model it as a long type?

Reply:
Yes. IMO wait time should be long, instead of timestamp (milliseconds since epoch)

Reply:
SGTM